### DISCUSSION TITLE
Documentation for creating custom middleware?

### DISCUSSION THREAD

**Discussion by aminert** [pmndrs/zustand#692] [Q&A] [answered]
Hi,

Are there any official documentation or best practices on how to create custom middleware? 

Thanks in advance!

**Comment by dai-shi** [comment, marked as answer]
Unfortunately, there are no docs for developing custom middleware. Better to learn from the code. It's not difficult in JS, but typing is extremely hard. #662 might change types in the future.

**Reply by aminert** [reply]
Got it, thanks. The Log middleware example was indeed easy to follow.
